core/src/actors/chain_manager/mod.rs

```diff
@@ -148,15 +148,20 @@ impl ChainManager {
             .collect()
     }
 
-    /// Method to update utxo set
-    fn update_utxo_set(&self, block: Block) -> HashMap<OutputPointer, Output> {
+    /// Method to update UTXO set
+    fn update_utxo_and_transactions_pool(
+        &self,
+        block: Block,
+    ) -> (HashMap<OutputPointer, Output>, TransactionsPool) {
         // Create a copy of the state "unspent_outputs_pool"
         let mut utxo_set = self.unspent_outputs_pool.clone();
+        let mut txn_pool = self.transactions_pool.clone();
```

Comment:
This operation will become inefficient very fast since the transactions pool might become quite large IMHO. I think we should either:
- modify the pool directly (without cloning)
- return a collection of the transactions to be deleted

@aesedepece what do you think?